server/implementation/src/main/java/io/smallrye/graphql/execution/context/SmallRyeContext.java

```diff
@@ -46,407 +23,213 @@
-        }
-        return null;
+    public void setPath(String path) {
+        this.path = path;
     }
 
     @Override
-    public List<String> getRequestedOperationTypes() {
-        List<String> allRequestedTypes = new ArrayList<>();
-
-        if (documentSupplier != null) {
-            Document document = documentSupplier.get();
-            List<OperationDefinition> definitions = document.getDefinitionsOfType(OperationDefinition.class);
-            for (OperationDefinition definition : definitions) {
-                String operationType = getOperationTypeFromDefinition(definition);
-                if (!allRequestedTypes.contains(operationType)) {
-                    allRequestedTypes.add(operationType);
-                }
-            }
-        }
-        return allRequestedTypes;
+    public JsonArray getSelectedFields() {
+        return this.selectedFields;
     }
 
-    @Override
-    public Optional<String> getParentTypeName() {
-        if (dfe != null) {
-            return getName(dfe.getParentType());
-        }
-        return Optional.empty();
+    public void setSelectedFields(JsonArray selectedFields) {
+        this.selectedFields = selectedFields;
     }
 
     @Override
-    public Optional<String> getOperationName() {
-        // Check if it's set in the request
-        if (hasOperationName()) {
-            String fromRequest = jsonObject.getString(OPERATION_NAME);
-            if (fromRequest != null && !fromRequest.isEmpty()) {
-                return Optional.of(fromRequest);
-            }
-        }
-
-        // Else try the input
-        if (executionInput != null && executionInput.getOperationName() != null
-                && !executionInput.getOperationName().isEmpty()) {
-            return Optional.of(executionInput.getOperationName());
-        }
-
-        // Else try DataFetchingEnvironment
-        if (dfe != null && dfe.getOperationDefinition() != null && dfe.getOperationDefinition().getName() != null
-                && !dfe.getOperationDefinition().getName().isEmpty()) {
-            return Optional.of(dfe.getOperationDefinition().getName());
-        }
-
-        return Optional.empty();
+    public JsonArray getSelectedAndSourceFields() {
+        return selectedAndSourceFields;
     }
 
-    @Override
-    public <K> boolean hasMetaData(K key) {
-        if (dfe != null) {
-            return dfe.getGraphQlContext().hasKey(key);
-        }
-        return false;
+    public void setSelectedAndSourceFields(JsonArray selectedAndSourceFields) {
+        this.selectedAndSourceFields = selectedAndSourceFields;
     }
 
     @Override
-    public <K, V> V getMetaData(K key) {
-        if (dfe != null) {
-            return dfe.getGraphQlContext().get(key);
-        }
-        return null;
+    public String getOperationType() {
+        return this.operationType;
     }
 
-    @Override
-    public <K, V> void putMetaData(K key, V value) {
```

Comment:
@jmartisk - see here, in the current version, the adding of metadata is not new...